lib/route/route.js

```diff
@@ -679,6 +679,32 @@ const initRoutes = (pages) => {
   // serviceDataTimestamp = getTimestamp()
 }
 
+function asyncRoute (middleware) {
+  function callNext ({withError, res, next}) {
```

Comment:
I don't know why but the callNext method stumped me and made me confused as to where `req` had disappeared to.

Not sure it's not clearer to skip the callNext

```js
function asyncRoute (middleware) {
  return function wrappedRoute (req, res, next) {
    middleware(req, res)
      .then(
        setTimeout(() => {
          if (!res.headersSent) {
            next()
          }
        }, 0))
      .catch((err) => {
        setTimeout(() => {
          if (!res.headersSent) {
            next(err)
          }
        }, 0))
  }
}
```

or even chain through to a final `then`

```js
function asyncRoute (middleware) {
  return function wrappedRoute (req, res, next) {
    middleware(req, res)
      .then(() => {})
      .catch(err => err)
      .then(err => {
        // Express does not set headersSent immediately
        setTimeout(() => {
          if (!res.headersSent) {
            next(err)
          }
        }, 0))
      })
  }
}
```

Is there any reason to use 1 rather than 0 for the timeout?

Reply:
Have inlined the function now.

Have moved to `nextTick`.

> Is there any reason to use 1 rather than 0 for the timeout?

Wanted to do that to avoid a potential race condition and stay on the same side, until I understood exactly what was causing the delay

Reply:
Cool - nextTick obviously not a hack ;) and the inlining reads much easier

Reply:
👍